iothub/service/src/Common/Exceptions/DeviceAlreadyExistsException.cs

```diff
@@ -6,30 +6,62 @@
     public sealed class DeviceAlreadyExistsException : IotHubException
     {
+        /// <summary>
+        /// Creates an instance of <see cref="DeviceAlreadyExistsException"/> with the specified device Id, an empty error message
+        /// and marks it as non-transient.
+        /// </summary>
+        /// <param name="deviceId">The Id of the device that is already registered on IoT Hub.</param>
         public DeviceAlreadyExistsException(string deviceId)
             : this(deviceId, string.Empty)
         {
         }
 
+        /// <summary>
+        /// Creates an instance of <see cref="DeviceAlreadyExistsException"/> with the specified device Id and the service returned tracking Id
+        /// associated with this particular error, and marks it as non-transient.
+        /// </summary>
+        /// <param name="deviceId">The Id of the device that is already registered on IoT Hub.</param>
+        /// <param name="trackingId">The service returned tracking Id associated with this particular error.</param>
         public DeviceAlreadyExistsException(string deviceId, string trackingId)
-            : base("Device {0} already registered".FormatInvariant(deviceId), trackingId)
+            : base($"Device {deviceId} already registered", trackingId)
         {
         }
 
+        /// <summary>
+        /// Creates an instance of <see cref="DeviceAlreadyExistsException"/> with a specified error message and
+        /// a reference to the inner exception that caused this exception, and marks it as non-transient.
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
         public DeviceAlreadyExistsException(string message, Exception innerException)
             : base(message, innerException)
         {
         }
 
+        /// <summary>
+        /// Creates an instance of <see cref="DeviceAlreadyExistsException"/> with a specified <see cref="ErrorCode"/>, error message and an
+        /// optional reference to the inner exception that caused this exception. This exception is marked as non-transient.
+        /// </summary>
+        /// <param name="code">The <see cref="ErrorCode"/> associated with the error.</param>
+        /// <param name="message">The message that describes the error.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
         public DeviceAlreadyExistsException(ErrorCode code, string message, Exception innerException = null)
             : base(code, message, innerException)
         {
         }
 
-        public DeviceAlreadyExistsException(SerializationInfo info, StreamingContext context)
+        internal DeviceAlreadyExistsException()
+            : base()
+        {
+        }
+
+        private DeviceAlreadyExistsException(SerializationInfo info, StreamingContext context)
```

Comment:
Since this exception is a sealed type, the serialization constructor is declared as private: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2229?view=vs-2019#cause